### ISSUE TITLE
If timber is imported but not installed, metadata is always appended

### ISSUE THREAD

**Issue by ariamckinley** [vectordotdev/timber-node#53]
This issue is occurring because the `console` patch is not installed explicitly by the user. It's immediately patched as soon as timber is imported.

So if an application is setup to conditionally install the timber transport like so:

```js
const timber = require('timber');

if (process.env === 'production' && process.env.TIMBER_API_KEY) {
  const transport = new timber.transports.HTTPS(process.env.TIMBER_API_KEY);

  timber.install(transport);
}
```

The logs produced while in a non-production environment will __always__ have metadata appended to them.

This could be fixed by changing the behavior of the library to make patching `console` explicit. This would make the user responsible for patching `console` themselves; which could be seen as both a pro or a con depending on the user's needs.

If this were done, the example above would end up looking like:

```js
const timber = require('timber');

if (process.env === 'production' && process.env.TIMBER_API_KEY) {
  const transport = new timber.transports.HTTPS(process.env.TIMBER_API_KEY);
  console = timber.console;

  timber.install(transport);
}
```

The benefit is that there is no "magic" happening when importing timber. Your logs won't get augmented using `console.log` unless you explicitly override `console` with our patched `console`.

The drawback is that it's one more step for the end-user to use the lib. But it may be worth it for the sake of making everything explicit. There would be no confusion as to why their logs are getting modified before even using the `install()` function.

**Comment by binarylogic** [comment]
Is there no way the `timber.install(transport);` could handle this? Ex:

```
console = timber.install(transport);
```

**Comment by ariamckinley** [comment]
Yeah we can return the console patch from the install function. What I was mainly getting at was that it would have to be explicitly patched by the user. We can definitely do that for convenience sake

If anybody uses the `attach()` functions instead of `install()` though, they'll need to patch the console like in my original example. It would look something like:

```js
timber.attach([ httpTransport, fileTransport ], process.stdout);
timber.attach([ httpTransport, fileTransport ], process.stderr);
console = timber.console;
```

**Comment by ariamckinley** [comment]
But the majority of people will be using `install()`, so it would save them a step.

**Comment by ariamckinley** [comment]
Closed via f86ece3b3b4447b38b1aebb57bde1cc06a48b6b7
